VM/vm_params2vm.py

```diff
@@ -0,0 +1,302 @@
+    )
+
+    (vm_working_dir, nzvm_cfg_path, _) = temp_paths(ptemp)
+
+    with open(vm_params_path, "r") as f:
+        vm_params_dict = yaml.load(f, Loader=yaml.SafeLoader)
+
+    if vm_params_dict is not None:
+        # saves nzvm.cfg
+        save_nzvm_cfg(nzvm_cfg_path, vm_params_dict, vm_working_dir, logger=logger)
+        # makes a copy of vm_params.yaml
+        copyfile(vm_params_path, os.path.join(ptemp, os.path.basename(vm_params_path)))
+        # run the actual generation
+        gen_vm(out_dir, ptemp, vm_threads=vm_threads, logger=logger)
+    else:
+        logger.debug("vm_params.yaml has no data for VM generation")
+
+    vm_params_dict["out_dir"] = out_dir
+
+    # working dir cleanup, return info about VM
+    logger.debug("Cleaning up temp directory {}".format(ptemp))
+    rmtree(ptemp)
+    return vm_params_dict
+
+
+def load_msgs_vm_params(args, logger: Logger = qclogging.get_basic_logger()):
+
+    msgs = []
+    logger.debug("Checking VM params file: {}".format(args.vm_params_path))
+
+    if not os.path.exists(args.vm_params_path):
+        raise FileNotFoundError(args.vm_params_path)
+
+    msgs.append(
+        (
+            args.name,
+            args.vm_params_path,
+            args.out_dir,
+            args.vm_threads,
+            qclogging.get_realisation_logger(logger, args.name).name,
+        )
+    )
+    return msgs
+
+
+def load_args(logger: Logger = qclogging.get_basic_logger()):
+
+    parser = ArgumentParser()
+    arg = parser.add_argument
+
+    arg("name", help="Name of the fault")
+
+    arg(
+        "vm_params_path",
+        help="path to vm_params.yaml",
+    )
+
+    arg("vm_root_dir", help="root directory to place VM files eg. Data/VMs")
+
+    arg("-n", "--nproc", help="number of processes", type=int, default=1)
+    arg(
+        "-t",
+        "--vm_threads",
+        "--threads",
+        help="number of threads for the VM generation",
+        type=int,
+        default=1,
+    )
+    arg("--vm-version", help="velocity model version to generate", default="1.65")
+    arg(
+        "--vm-topo",
+        help="topo_type parameter for velocity model generation",
+        default="BULLDOZED",
+    )
+
+    args = parser.parse_args()
+    args.vm_params_path = os.path.abspath(args.vm_params_path)
+    args.vm_root_dir = os.path.abspath(args.vm_root_dir)
+    return args
+
+
+if __name__ == "__main__":
+
+    logger = qclogging.get_logger("vm_params2vm")
+    qclogging.add_general_file_handler(
+        logger, os.path.join(os.getcwd(), "vm_params2vm_log.txt")
+    )
+    args = load_args(logger=logger)
+
+    # prepare to run
+    args.out_dir = os.path.join(args.vm_root_dir, args.name)
+    if not os.path.isdir(args.out_dir):
+        logger.debug(
+            "VM output directory {} does not exist. Creating it now.".format(
+                args.out_dir
+            )
+        )
+
+        os.makedirs(args.out_dir)
+
+    msg_list = load_msgs_vm_params(args, logger=logger)
```

Comment:
Shouldn't pass args around. Especially as this script should only have ~3 arguments